tests/NHapi.NUnit/Parser/PipeParserTests.cs

```diff
@@ -308,5 +309,96 @@ public void MoreGreedyMode()
                 }
             }
         }
+
+        /// <summary>
+        /// The folllwing 4 tests are ported from hapi
+        /// https://github.com/hapifhir/hapi-hl7v2/blob/master/hapi-examples/src/main/java/ca/uhn/hl7v2/examples/ParseInvalidObx2Values.java.
+        /// </summary>
+        [Test]
+        public void Obx5DataTypeIsSetFromObx2_WhenObx2IsEmpty_Hl7ExceptionIsThrown()
+        {
+            var message =
+                "MSH|^~\\&|XPress Arrival||||200610120839||ORU^R01|EBzH1711114101206|P|2.3|||AL|||ASCII\r"
+              + "PID|1||1711114||Appt^Test||19720501||||||||||||001020006\r"
+              + "ORC|||||F\r"
+              + "OBR|1|||ehipack^eHippa Acknowlegment|||200610120839|||||||||00002^eProvider^Electronic|||||||||F\r"
+              + "OBX|1||||STValue||||||F\r";
+
+            var parser = new PipeParser();
+
+            var exception = Assert.Throws<HL7Exception>(() => parser.Parse(message));
+
+            Assert.AreEqual(
+                "OBX-5 is valued, but OBX-2 is not.  A datatype for OBX-5 must be specified using OBX-2.",
+                exception.Message);
```

Comment:
Instead of checking the exact exception message text, can we check the `ErrorCode` or similar? My concern is that relying on the exact text could be fragile.

Reply:
I could use the error code but there are other places in the `PipeParser` pipeline that throw hl7 exceptions with the same error code. This test is looking for a very specific exception, which is why I check the message.

I do understand your reasoning though but I think in this case highly unlikely to change (It's been the same since at least 2014 when the source code moved to GitHub), and even if it did it would be a small effort to update the unit test.